src/cmd/linuxkit/run_virtualizationframework_darwin.go

```diff
@@ -0,0 +1,324 @@
+	networking := flags.String("networking", virtualizationNetworkingDefault, "Networking mode. Valid options are 'default', 'vmnet' and 'none'. 'vmnet' uses the Apple vmnet framework. 'none' disables networking.`")
+
+	kernelBoot := flags.Bool("kernel", false, "Boot image is kernel+initrd+cmdline 'path'-kernel/-initrd/-cmdline")
+
+	if err := flags.Parse(args); err != nil {
+		log.Fatal("Unable to parse args")
+	}
+	remArgs := flags.Args()
+	if len(remArgs) == 0 {
+		fmt.Println("Please specify the prefix to the image to boot")
+		flags.Usage()
+		os.Exit(1)
+	}
+	path := remArgs[0]
+	prefix := path
+
+	_, err := os.Stat(path + "-kernel")
+	statKernel := err == nil
+
+	var isoPaths []string
+
+	// Default to kernel+initrd
+	if !statKernel {
+		log.Fatalf("Cannot find kernel file: %s", path+"-kernel")
+	}
+	_, err = os.Stat(path + "-initrd.img")
+	statInitrd := err == nil
+	if !statInitrd {
+		log.Fatalf("Cannot find initrd file (%s): %v", path+"-initrd.img", err)
+	}
+	*kernelBoot = true
+
+	metadataPaths, err := CreateMetadataISO(*state, *data, *dataPath)
+	if err != nil {
+		log.Fatalf("%v", err)
+	}
+	isoPaths = append(isoPaths, metadataPaths...)
+
+	// We should generate new UUID, otherwise /sys/class/dmi/id/product_uuid is identical on all VMs.
+	// but it is not clear if support is there in VF, or if it is built-in
+
+	// Run
+
+	cmdlineBytes, err := ioutil.ReadFile(prefix + "-cmdline")
+	if err != nil {
+		log.Fatalf("Cannot open cmdline file: %v", err)
+	}
+	// must have hvc0 as console for vf
+	kernelCommandLineArguments := strings.Split(string(cmdlineBytes), " ")
+
+	// Use the first virtio console device as system console.
+	//"console=hvc0",
+	// Stop in the initial ramdisk before attempting to transition to
+	// the root file system.
+	//"root=/dev/vda",
+	kernelCommandLineArguments = append(kernelCommandLineArguments, "console=hvc0")
+
+	vmlinuz := prefix + "-kernel"
+	initrd := prefix + "-initrd.img"
+
+	vmlinuzFile := vmlinuz
+	// need to check if it is gzipped, and, if so, gunzip it
+	filetype, err := checkFileType(vmlinuz)
+	if err != nil {
+		log.Fatalf("unable to check kernel file type at %s: %v", vmlinuz, err)
+	}
+
+	if filetype == "application/x-gzip" {
+		vmlinuzUncompressed := fmt.Sprintf("%s-uncompressed", vmlinuz)
+		// gzipped kernel, we load it into memory, unzip it, and pass it
+		f, err := os.Open(vmlinuz)
+		if err != nil {
+			log.Fatalf("unable to read kernel file %s: %v", vmlinuz, err)
+		}
+		defer f.Close()
+		r, err := gzip.NewReader(f)
+		if err != nil {
+			log.Fatalf("unable to read from file %s: %v", vmlinuz, err)
+		}
+		defer r.Close()
+
+		writer, err := os.Create(vmlinuzUncompressed)
+		if err != nil {
+			log.Fatalf("unable to create uncompressed kernel file %s: %v", vmlinuzUncompressed, err)
+		}
+		defer writer.Close()
+
+		if _, err = io.Copy(writer, r); err != nil {
+			log.Fatalf("unable to uncompressed kernel file to %s: %v", vmlinuzUncompressed, err)
```

Comment:
`s/uncompressed/uncompress/g`

Reply:
Actually, it probably should be "decompress". I will fix it.